src/main/java/com/aws/greengrass/detector/IpDetectorService.java

```diff
@@ -63,9 +67,29 @@ public void startup() throws InterruptedException {
     @Override
     public void shutdown() throws InterruptedException {
         logger.atInfo().log("Stop IP detection task");
-        if (future != null) {
-            future.cancel(true);
+        if (executorServiceFuture != null) {
+            executorServiceFuture.cancel(true);
         }
         super.shutdown();
     }
+
+    private void subscribeToConfigs() {
+        if (portConfig == null) {
+            portConfig = this.config.lookup(KernelConfigResolver.CONFIGURATION_CONFIG_KEY, PORT);
+            if (portConfig != null) {
+                subscribeToMqttPortConfig();
+                logger.atInfo().log("Successfully subscribed to new port config");
+            }
+        }
+    }
+
+    private void subscribeToMqttPortConfig() {
+        portConfig.subscribe((whatHappened, node) -> {
+            Integer port = Coerce.toInt(portConfig);
+            if (port != null) {
+                ipDetectorConfig.setMqttPort(port);
+                ipDetectorManager.updateIps(ipDetectorConfig);
+            }
+        });
+    }
```

Comment:
This file doesn't need any changes (except fixing the typo above). The config object reads config